<a href="https://colab.research.google.com/github/sebastianayalagomez03/PRIMER-CORTE/blob/main/Copia_de_Taller1_Analisis_de_Redes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## ADDIDAS ANALISIS

In [16]:
!wget https://github.com/prayagpadwal/Adidas-Sales-Analysis/raw/main/Adidas%20US%20Sales%20Datasets.xlsx

--2025-09-09 01:01:53--  https://github.com/prayagpadwal/Adidas-Sales-Analysis/raw/main/Adidas%20US%20Sales%20Datasets.xlsx
Resolving github.com (github.com)... 20.27.177.113
Connecting to github.com (github.com)|20.27.177.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/prayagpadwal/Adidas-Sales-Analysis/main/Adidas%20US%20Sales%20Datasets.xlsx [following]
--2025-09-09 01:01:54--  https://raw.githubusercontent.com/prayagpadwal/Adidas-Sales-Analysis/main/Adidas%20US%20Sales%20Datasets.xlsx
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 693038 (677K) [application/octet-stream]
Saving to: ‘Adidas US Sales Datasets.xlsx.1’

Adidas US Sales Dat 100%[===================>] 676.79K  2.78MB/s    in

In [17]:
import pandas as pd
data = pd.read_excel("Adidas US Sales Datasets.xlsx")

In [18]:
data.head(3)

Retailer  Retailer ID Invoice Date     Region     State      City  \
0  Foot Locker      1185732   2020-01-01  Northeast  New York  New York   
1  Foot Locker      1185732   2020-01-02  Northeast  New York  New York   
2  Foot Locker      1185732   2020-01-03  Northeast  New York  New York   

                   Product  Price per Unit  Units Sold  Total Sales  \
0    Men's Street Footwear            50.0        1200     600000.0   
1  Men's Athletic Footwear            50.0        1000     500000.0   
2  Women's Street Footwear            40.0        1000     400000.0   

   Operating Profit  Operating Margin Sales Method  
0          300000.0              0.50     In-store  
1          150000.0              0.30     In-store  
2          140000.0              0.35     In-store

0. Cuantas filas y columnas tiene la tabla?

In [19]:
data.shape

(9648, 13)

0.1 Cual es la unidad de observación de la tabla?



```
La unidad de observación es cada registro de venta de Adidas en EE. UU., con sus valores de ventas, ganancia y margen.
```



1. Cuales fueron las ventas totales?

In [20]:
data["Total Sales"].sum()

np.float64(899902125.0)

1.1 Cuanto fue la ganancia operacional total?

In [21]:
data["Operating Profit"].sum()

np.float64(332134761.45000005)

1.2 Cuanto fue el margen promedio total ?

In [22]:
data["Operating Margin"].mean()

np.float64(0.42299129353233833)

:1.3 Por que el margen promedio y no el total. Explique con palabras

El margen es un porcentaje, no se puede sumar. Por eso se usa el promedio, que refleja mejor la rentabilidad global.



2. Cuales fueron las ventas totales por Retailer?

In [23]:
data.groupby("Retailer")["Total Sales"].sum()

Retailer
Amazon            77698912.0
Foot Locker      220094720.0
Kohl's           102114753.0
Sports Direct    182470997.0
Walmart           74558410.0
West Gear        242964333.0
Name: Total Sales, dtype: float64

2.1 Cual fue la ganancia operacional total por retailer?

In [24]:
data.groupby("Retailer")["Operating Profit"].sum()

Retailer
Amazon           28818503.31
Foot Locker      80722124.81
Kohl's           36811252.58
Sports Direct    74332954.96
Walmart          25782052.61
West Gear        85667873.18
Name: Operating Profit, dtype: float64

2.3 Tomando los dos resultados de arriba: El ganancia operacional total y las ventas totales divida ambos resultados. Le tiene que dar el margen total de cada Retailer.

Amazon	      28818503.31 / 77698912.0 = 0.3708
Foot Locker	  80722124.81 / 220094720.0 = 0.3667
Kohl's	      36811252.58 / 122114753.0 = 0.3014
Sports Direct	74332954.96 / 182470997.0 = 0.4073
Walmart	      25782052.61 / 74558410.0 = 0.3457
West Gear	    85667873.18 / 242964333.0 = 0.3525


2.4 Cual fue el margen operacional promedio por retailer?

In [25]:
data.groupby("Retailer")["Operating Margin"].mean()

Retailer
Amazon           0.417871
Foot Locker      0.417933
Kohl's           0.419291
Sports Direct    0.444892
Walmart          0.406534
West Gear        0.417856
Name: Operating Margin, dtype: float64

Los resultados son distintos porque en 2.3 se calcula un margen ponderado por ventas, y en 2.4 un promedio simple. El primero refleja la rentabilidad real total de cada retailer, mientras que el segundo suaviza las diferencias al no considerar el peso de las ventas.

3.1 Retailer que mas unidades totales vende? Saque cuanto vende en unidades cada retailer y busque el mayor

In [26]:
data.groupby("Retailer")["Units Sold"].sum().sort_values(ascending=False)

Retailer
West Gear        625262
Foot Locker      604369
Sports Direct    557640
Kohl's           287375
Walmart          206225
Amazon           197990
Name: Units Sold, dtype: int64

3.2 Cual fue el producto que mas unidades totales vendió?

In [27]:
data.groupby("Product")["Units Sold"].sum().sort_values(ascending=False)

Product
Men's Street Footwear        593320
Men's Athletic Footwear      435526
Women's Apparel              433827
Women's Street Footwear      392269
Women's Athletic Footwear    317236
Men's Apparel                306683
Name: Units Sold, dtype: int64

3.3 Cual es el metodo de venta que mas ventas totales tuvo y más ventas promedio

In [28]:
data.groupby("Sales Method")["Total Sales"].sum().sort_values(ascending=False)

Sales Method
In-store    356643750.0
Outlet      295585493.0
Online      247672882.0
Name: Total Sales, dtype: float64

In [29]:
data.groupby("Sales Method")["Total Sales"].mean().sort_values(ascending=False)

Sales Method
In-store    204967.672414
Outlet       97908.411063
Online       50659.210882
Name: Total Sales, dtype: float64

3.4 Cual Fue el estado que más ventas totales tuvo ?

In [30]:
data.groupby("State")["Total Sales"].sum().sort_values(ascending=False)

State
New York          64229039.0
California        60174133.0
Florida           59283714.0
Texas             46359746.0
South Carolina    29285637.0
Washington        26330718.0
North Carolina    23956531.0
Louisiana         23750781.0
Hawaii            22282457.0
Virginia          21575040.0
Oregon            21349674.0
Colorado          20996536.0
Nevada            20858509.0
New Mexico        19865016.0
Idaho             19276878.0
Georgia           18997466.0
Michigan          18625433.0
Wyoming           18577517.0
Ohio              18484583.0
Tennessee         18067440.0
Alabama           17633424.0
New Hampshire     16411667.0
Arizona           15782221.0
Montana           15710886.0
Mississippi       15591709.0
Alaska            14753103.0
Vermont           14352923.0
Arkansas          12639347.0
Delaware          12298412.0
Connecticut       11573448.0
Massachusetts     11076810.0
West Virginia     10689160.0
Oklahoma          10620734.0
Pennsylvania      10357519.0
Kentucky          10072848.0
Kansas             9972864.0
New Jersey         9934246.0
Illinois           9797488.0
Utah               9696420.0
Missouri           9683265.0
Maine              9195978.0
Indiana            8836198.0
South Dakota       8495576.0
Rhode Island       8446806.0
Maryland           7758059.0
North Dakota       7735580.0
Wisconsin          7727469.0
Iowa               7424011.0
Minnesota          7378068.0
Nebraska           5929038.0
Name: Total Sales, dtype: float64

4. Teniendo en cuenta los resultados de arriba: 1. Tome el Retailer que vendio
mas 2. El estado que vendio mas 3.el producto que vendio mas y el metodo de ventas que vendio más 4. Filtre y genere un dataframe con ellos 5. llamelo como un nuevo dataframe

In [34]:
# 1. Retailer que más vendió
top_retailer = data.groupby("Retailer")["Units Sold"].sum().idxmax()

# 2. Estado que más vendió
top_state = data.groupby("State")["Units Sold"].sum().idxmax()

# 3. Producto que más vendió
top_product = data.groupby("Product")["Units Sold"].sum().idxmax()

# 4. Método de venta que más vendió
top_method = data.groupby("Sales Method")["Units Sold"].sum().idxmax()

# 5. Filtrar dataframe con esos valores
nuevo_df = data[
    (data["Retailer"] == top_retailer) &
    (data["State"] == top_state) &
    (data["Product"] == top_product) &
    (data["Sales Method"] == top_method)
]

# Mostrar el nuevo dataframe
nuevo_df


Retailer  Retailer ID Invoice Date     Region     State    City  \
5331  West Gear      1185732   2021-03-08  Northeast  New York  Albany   
5337  West Gear      1185732   2021-04-09  Northeast  New York  Albany   
5343  West Gear      1185732   2021-05-08  Northeast  New York  Albany   
5349  West Gear      1185732   2021-06-10  Northeast  New York  Albany   
5355  West Gear      1185732   2021-07-08  Northeast  New York  Albany   
5361  West Gear      1185732   2021-08-09  Northeast  New York  Albany   
5367  West Gear      1185732   2021-09-08  Northeast  New York  Albany   
5373  West Gear      1185732   2021-10-10  Northeast  New York  Albany   
5379  West Gear      1185732   2021-11-09  Northeast  New York  Albany   
5385  West Gear      1185732   2021-12-08  Northeast  New York  Albany   

                    Product  Price per Unit  Units Sold  Total Sales  \
5331  Men's Street Footwear            44.0         252      11088.0   
5337  Men's Street Footwear            42.0         248      10416.0   
5343  Men's Street Footwear            50.0         261      13050.0   
5349  Men's Street Footwear            59.0         263      15517.0   
5355  Men's Street Footwear            59.0         247      14573.0   
5361  Men's Street Footwear            62.0         250      15500.0   
5367  Men's Street Footwear            50.0         238      11900.0   
5373  Men's Street Footwear            57.0         208      11856.0   
5379  Men's Street Footwear            56.0         217      12152.0   
5385  Men's Street Footwear            59.0         245      14455.0   

      Operating Profit  Operating Margin Sales Method  
5331           6541.92              0.59       Online  
5337           5832.96              0.56       Online  
5343           7830.00              0.60       Online  
5349           8689.52              0.56       Online  
5355           8015.15              0.55       Online  
5361           8680.00              0.56       Online  
5367           7021.00              0.59       Online  
5373           6757.92              0.57       Online  
5379           6683.60              0.55       Online  
5385           8383.90              0.58       Online